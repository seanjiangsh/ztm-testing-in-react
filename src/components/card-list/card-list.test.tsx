import { shallow } from "enzyme";

import CardList, { CardListProps } from "./card-list.component";

const mockMonsters: CardListProps = {
  monsters: [
    { id: 1, name: "Test Monster 1", email: "monster1@gmail.com" },
    { id: 2, name: "Test Monster 2", email: "monster2@gmail.com" },
  ],
};
test("expect to render CardList component", () => {
  const comp = shallow(<CardList {...mockMonsters} />);
  console.log(comp);
  expect.assertions(1);
  expect(comp.length).toEqual(1);
});
