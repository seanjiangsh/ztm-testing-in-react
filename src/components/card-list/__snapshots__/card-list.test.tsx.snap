// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
<div
  className="card-list"
>
  <Card
    key="testMonster1"
    monster={
      Object {
        "email": "monster1@gmail.com",
        "id": "testMonster1",
        "name": "Test Monster 1",
      }
    }
  />
  <Card
    key="testMonster2"
    monster={
      Object {
        "email": "monster2@gmail.com",
        "id": "testMonster2",
        "name": "Test Monster 2",
      }
    }
  />
</div>
`;
